### PR TITLE
Add support for new languages.

### PR DESCRIPTION
**Pull request by darewreck54**
image

Add support for the languages specified in the table above.

### CODE REVIEW

**Comment by darewreck54** [comment]
I notice that for the acronym for the language are only two characters.  Shouldn't it be 4 characters?

**Comment by ljharb** [comment]
I believe the overarching language is two characters, and the locale-specific language is 4.

hr is Croatian, which follows the same pluralization rules as russian.

**Comment by darewreck54** [comment]
Am i reading this incorrectly...They seem ddifferent

http://www.unicode.org/cldr/charts/33/supplemental/language_plural_rules.html

image

**Comment by ljharb** [comment]
It's always possible we have a bug :-) if you want to open a separate PR with test cases for croatian, then we can get that fix in quickly.

**Comment by darewreck54** [comment]
i'll update this PR with it.  Also should we be using the locale-specific code?

**Comment by ljharb** [comment]
I'd prefer it be a separate PR - this one is a semver-minor, adding new features - bugfixes are semver-patch and I'd like to get those in separately.

**Comment by darewreck54** [comment]
Any guidance on the language acronyms?  Is pluralization only based on the language code?  Is there no variation between different region(locale)?

https://www.science.co.il/language/Locale-codes.php

It looks like the current code only handles the language code and not really the locale id?

**Comment by ljharb** [comment]
It supports both: https://github.com/airbnb/polyglot.js/blob/master/index.js#L86-L87

**Comment by darewreck54** [comment]
If it supports both, i'm guessing it will default to the language code.  Does that present any potential issues when a user uses it the method?  For example, they mistakenly pass in the language code vs. the local which might result in unexpected results?

**Comment by darewreck54** [comment]
@ljharb 

I've removed the Croatian fix to another PR.  How long do you think this PR will take to get approved? ETA?

Thanks,
Derek

**Comment by ljharb** [comment]
Thanks for filing #114 :-)

As for this one, I'm waiting on some internal l10n folks to review this PR - I'm hoping to merge it early next week.

**Comment by darewreck54** [comment]
has your team been able to verify the new languages?

**Comment by ljharb** [comment]
Since we don't support these languages, it turns out our team won't be able to verify them. I'll rebase this, and give it a review myself.

File: index.js [hunk 1]
```diff
@@ -72,22 +80,34 @@ var pluralTypes = {
     if (n === 1) { return 0; }
     return n % 10 >= 2 && n % 10 <= 4 && (n % 100 < 10 || n % 100 >= 20) ? 1 : 2;
   },
-  icelandic: function (n) { return (n % 10 !== 1 || n % 100 === 11) ? 1 : 0; }
+  icelandic: function (n) { return (n % 10 !== 1 || n % 100 === 11) ? 1 : 0; },
+  slovenian: function (n) {
+    if (n === 1 || n % 100 === 1) {
```
Comment (ljharb) [inline index.js]: This seems incorrect; it's numbers ending in *01* - so, `n === 1 || (n > 100 && n % 100 === 1)`

File: index.js [hunk 2]
```diff
@@ -72,22 +80,34 @@ var pluralTypes = {
...
+      return 0;
+    } else if (n === 2 || n % 100 === 2) {
```
Comment (ljharb) [inline index.js]: same here; it's ending in 02, including 2 (not ending in 2)

File: index.js [hunk 3]
```diff
@@ -72,22 +80,34 @@ var pluralTypes = {
...
+      return 1;
+    } else if (n === 3 || n === 4 || n % 100 === 3 || n % 100 === 4) {
```
Comment (ljharb) [inline index.js]: also here

**Comment by ljharb** [comment]
It'd be great if you could rebase your branch on top of fe829c39638f4c26f109e48fe7953707a3dee505 :-)

File: index.js (hunk 1 above)
Comment (darewreck54) [reply index.js]: I add the fix, but i don't beleive a number can be 01...since i'm assuming your passing in a number vs. a string.

**Comment by darewreck54** [comment]
addressed all your comments

File: index.js (hunk 1 above)
Comment (ljharb) [reply index.js]: right - that's why the `n > 100`. because the only way a number can end in "01" is if it has three or more digits.

**Comment by darewreck54** [comment]
sorry i'm not following...

 `if (n % 100 === 1)) 
`
101%100= 1(true), 1001%100 = 1 (true), 12301%100 = 1 (true) 11%100 = 11 (false)

I've also added test to validate it as well.

As for separating russian and croatian (done in previous PR), does it make sense because it's already done for other languages such as chinese/korea/thai/etc.

**Pull request merged**